extensions/interactions/GraphInput/directives/oppia-response-graph-input.component.ts

```diff
@@ -58,6 +64,73 @@ export class ResponseGraphInput {
     ) as GraphAnswer;
     this.VERTEX_RADIUS = this.graphDetailService.VERTEX_RADIUS;
     this.EDGE_WIDTH = this.graphDetailService.EDGE_WIDTH;
+
+    this.MIN_MARGIN = this.graphDetailService.getMinMargin(this.graph);
+    this.minX = this.graphDetailService.getMinX(this.graph);
+    this.minY = this.graphDetailService.getMinY(this.graph);
+
+    this.reduceGraph();
+  }
+
+  removeWhiteSpace(): void {
+    // The first vertex should be at (MIN_MARGIN, MIN_MARGIN).
+    this.graph.vertices = this.graph.vertices.map(vertex => {
+      return {
+        x: vertex.x - this.minX + this.MIN_MARGIN,
+        y: vertex.y - this.minY + this.MIN_MARGIN,
+        label: vertex.label,
+      };
+    });
+  }
+
+  reduceGraph(): void {
+    this.removeWhiteSpace();
+
+    var scale = this.getScale();
+    if (scale === 1) {
+      return;
+    }
+
+    // If the scale is not 1, resize the graph with the scale factor.
+    this.graph.vertices = this.graph.vertices.map(vertex => {
+      return {
+        x: vertex.x * scale,
+        y: vertex.y * scale,
+        label: vertex.label,
+      };
+    });
+  }
+
+  getMaxX(): number {
+    return Math.max(...this.graph.vertices.map(vertex => vertex.x));
+  }
+
+  getMaxY(): number {
+    return Math.max(...this.graph.vertices.map(vertex => vertex.y));
+  }
+
+  // This function calculates the scale factor required to fit the graph.
```

Comment:
...to fit the graph within the display area.

Reply:
Done